Automattic/newspack-plugin
# feat(amp-plus): allow OneSignal script

### All Submissions:






### Changes proposed in this Pull Request:



Alternate approach from #417 to handling push notifications. Instead of making them work within AMP constraints, using AMP Plus mode.

### How to test the changes in this Pull Request:

1. Deactivate the [PWA plugin](https://wordpress.org/plugins/pwa), if active (see https://github.com/Automattic/newspack-plugin/issues/2104)
1. Register and create a new app at [app.onesignal.com](https://app.onesignal.com) – be sure to choose "WordPress" as the integration in the setup wizard
2. Install [OneSignal WP plugin](https://wordpress.org/plugins/onesignal-free-web-push-notifications/) (`onesignal-free-web-push-notifications`) and configure it
3. Ensure AMP plugin is active and AMP Plus is enabled (`NEWSPACK_AMP_PLUS_ENABLED` environment variable is set to `true`)
4. Visit the homepage in a new (non-incognito!) session
5. Observe a notification bell icon appear in the lower right corner
6. Click it to subscribe to notifications
7. From the OneSignal dashboard, observe a new user has been added (in Audience -> Users)
8. Send a new notification, observe that it works 

### Other information:
